django/db/backends/postgresql/base.py

```diff
@@ -329,9 +330,12 @@ def _nodb_cursor(self):
                     )
                     try:
                         with conn.cursor() as cursor:
+                            cursor_yielded = True
                             yield cursor
                     finally:
                         conn.close()
```

Comment:
Look at this in more details won't this open an extra cursor that never gets closed if `DATABASES` has more one `postgres` entries with a `connection.settings_dict['NAME'] != 'postgres'`?

It looks like we need to `break` after the first iteration otherwise we'll hit a [`RuntimeError("generator didn't stop")` on `__exit__`](https://github.com/python/cpython/blob/9eb11a139fac5514d8456626806a68b3e3b7eafb/Lib/contextlib.py#L144))

```python
for connection in connections.all():
    if connection.vendor == 'postgresql' and connection.settings_dict['NAME'] != 'postgres':
        conn = self.__class__(
            {**self.settings_dict, 'NAME': connection.settings_dict['NAME']},
            alias=self.alias,
        )
        try:
            with conn.cursor() as cursor:
                yield cursor
        finally:
            conn.close()
        break
else:
    raise

Reply:
Good catch :dart: , thanks. I pushed edits and extra tests.

Reply:
@charettes, what if I have 3 connection configuration for Postgres and only third correct? Maybe collect all exceptions and do break only if one of there correct? Otherwise, raise a multi error. What do you think?

Reply:
@charettes then you'll get a crash on the first attempt at creating a connection. The point of this patch is not to support all possible misconfiguration of `DATABASES` but to make sure an adequate exception is surfaced.